janusgraph-core/src/main/java/org/janusgraph/graphdb/query/index/DefaultIndexSelector.java

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.query.index;
+
+import org.janusgraph.core.JanusGraphElement;
+import org.janusgraph.graphdb.database.IndexSerializer;
+import org.janusgraph.graphdb.internal.OrderList;
+import org.janusgraph.graphdb.query.condition.Condition;
+import org.janusgraph.graphdb.query.condition.MultiCondition;
+import org.janusgraph.graphdb.query.graph.JointIndexQuery;
+import org.janusgraph.graphdb.transaction.StandardJanusGraphTx;
+import org.janusgraph.graphdb.types.IndexType;
+import org.javatuples.Pair;
+
+import java.util.Set;
+
+/**
+ * @author Florian Grieskamp (Florian.Grieskamp@gdata.de)
+ */
+public class DefaultIndexSelector implements IndexSelectionStrategy {
+
+    /**
+     * Transaction which is used to gather settings of the graph
+     */
+    private StandardJanusGraphTx tx;
+
+    private IndexSelectionStrategy bruteForce;
+    private IndexSelectionStrategy approximation;
+
+    public DefaultIndexSelector(StandardJanusGraphTx tx) {
+        this.tx = tx;
+
+        this.bruteForce = new BruteForceIndexSelector();
+        this.approximation = new ApproximateIndexSelector();
+    }
+
+    @Override
+    public Pair<JointIndexQuery, Boolean> selectIndices(final Set<IndexType> indexCandidates, final MultiCondition<JanusGraphElement> conditions,
+                                                        final Set<Condition> coveredClauses, OrderList orders, IndexSerializer serializer) {
+        /* TODO: smarter optimization:
+        - use in-memory histograms to estimate selectivity of PredicateConditions and filter out low-selectivity ones
+                if they would result in an individual index call (better to filter afterwards in memory)
+        */
+        int bruteForceThreshold = tx.getGraph().getConfiguration().getIndexSelectThreshold();
```

Comment:
Can `bruteForceThreshold` be changed during different calls of this method? If no, could you provide `bruteForceThreshold` as a dependency of this class?

Reply:
Good point